src/map/clif.c

```diff
@@ -2378,23 +2378,37 @@ void clif_addcards2(unsigned short *cards, struct item* item) {
 }
 
 /**
- * Fills in RandomOptions(Bonuses) of items into the buffer
+ * Fills in ItemOptions(Bonuses) of items into the buffer
  *
- * Dummy datais used since this feature isn't supported yet (ITEM_RDM_OPT).
- * A maximum of 5 random options can be supported.
+ * A maximum of 5 item options can be supported.
  *
  * @param buf[in,out] The buffer to write to. The pointer must be valid and initialized.
  * @param item[in]    The source item.
  */
-void clif_add_random_options(unsigned char* buf, struct item* item)
-{
-	int i;
-	nullpo_retv(buf);
-	for (i = 0; i < 5; i++){
-		WBUFW(buf,i*5+0) = 0; // OptIndex
-		WBUFW(buf,i*5+2) = 0; // Value
-		WBUFB(buf,i*5+4) = 0; // Param1
+unsigned int clif_add_random_options(struct ItemOptions *buf, struct item *it)
```

Comment:
The problem here is that the HPM generators do not append the second const, I have not checked the scripts but `struct item *const it` would be generated as `struct item *constit`.